src/base/locale_helper.hpp

```diff
@@ -0,0 +1,208 @@
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ */
+
+/*
+ * locale_helper.hpp
+ */
+
+#ifndef _LOCALE_HELPER_HPP_
+#define _LOCALE_HELPER_HPP_
+
+#include "language_support.h"
+#include "intl_support.h"
+
+#include <locale>
+#include <codecvt>
+#include <string>
+
+/* TODO: need to make compatible fully between CUBRID Language support and cpp standard locale */
+
+namespace cublocale
+{
+  std::string get_lang_name (const LANG_COLLATION *lang_coll)
+  {
+    const char *lang_name = lang_coll->default_lang->lang_name;
+    std::string lang_str (lang_name);
+    return lang_str;
+  }
+
+  std::string get_codeset_name (const LANG_COLLATION *lang_coll)
+  {
+    const char *codeset_name = lang_get_codeset_name (lang_coll->codeset);
+    std::string codeset_str (codeset_name);
+    return codeset_str;
+  }
+
+  std::locale get_locale (const std::string &charset, const std::string &lang)
+  {
+    try
+      {
+	std::locale loc (lang + "." + charset);
+	return loc;
+      }
+    catch (std::exception &e)
+      {
+	// return the environment's default locale, locale name is not supported
+	assert (false);
+	return std::locale ("");
+      }
+  }
+
+  bool convert_to_wstring (std::wstring &out, const std::string &in, const LANG_COLLATION *lang_coll)
+  {
+    INTL_CODESET codeset = lang_coll->codeset;
+    bool is_success = false;
+
+    if (in.empty ())
+      {
+	// don't need to convert for empty string
+	out.clear ();
+	return true;
+      }
+
+    std::string utf8_str;
+    if (codeset != INTL_CODESET_UTF8)
+      {
+	std::string utf8_converted;
+	utf8_converted.resize (in.size() * INTL_CODESET_MULT (INTL_CODESET_UTF8));
+	std::string::pointer utf8_str_ptr = (char *) utf8_converted.data();
+
+	int conv_status = 0;
+	int conv_size = 0;
+	switch (codeset)
+	  {
+	  case INTL_CODESET_ISO88591:
+	    conv_status = intl_fast_iso88591_to_utf8 ((const unsigned char *) in.data(), in.size(),
+			  (unsigned char **) &utf8_str_ptr, &conv_size);
+	    break;
+	  case INTL_CODESET_KSC5601_EUC:
+	    conv_status = intl_euckr_to_utf8 ((const unsigned char *) in.data(), in.size(), (unsigned char **) &utf8_str_ptr,
+					      &conv_size);
+	    break;
+	  case INTL_CODESET_RAW_BYTES:
+	    intl_binary_to_utf8 ((const unsigned char *) in.data(), in.size(), (unsigned char **) &utf8_str_ptr,
+				 &conv_size);
+	    break;
+	  default:
+	    // unrecognized codeset
```

Comment:
I think explicit error message output is required for unsupported codesets.

Reply:
I don't think an explicit error is required. Even If new codeset is supported in CUBRID in the future, the assertion is enough to be checked by developers.

Reply:
i agree